makefiles/pseudomodules.inc.mk

```diff
@@ -102,6 +102,8 @@ PSEUDOMODULES += shell_hooks
 PSEUDOMODULES += slipdev_stdio
 PSEUDOMODULES += sock
 PSEUDOMODULES += sock_async
+PSEUDOMODULES += sock_aux_local
+PSEUDOMODULES += sock_aux_timestamp
```

Comment:
Not sure if this is a premature optimization or if this is even making a dent (we should compare binary size if this suggestion is accepted), but how about also adding a pseudo-module `sock_aux` that dependent on both. I.e. add

```makefile
ifneq (,$(filter sock_aux_%,$(USEMODULE)))
  USEMODULE += sock_aux
endif
```

So one can use `IS_USED(MODULE_SOCK_AUX)` in the code where applicable?

Reply:
I'm not sure if we really one two flavors of the recv and send functions to be implemented by the network stack. Currently, the non-aux flavors just call the aux flavors with `aux == NULL`.

The additional overhead currently is (in absence of any auxiliary data being used):

``` C
if (aux != NULL) {
    aux->flags = 0;
}
```

On most platforms this will be two instructions. If we indeed want to provide only one flavor of send and recv functions, those two instructions would make sure that user code never incorrectly assumes auxiliary data to be present (due to uninitialized `flags` member).

Reply:
Ok